### PR TITLE
Release workflow simplifications

### PR DESCRIPTION
**Pull request by trask**
Same as https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/6169

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -116,6 +129,8 @@ jobs:
           cat > /tmp/release-notes.txt << EOF
           This release targets the OpenTelemetry SDK $VERSION.
 
+          Note that all artifacts other than \`io.opentelemetry.javaagent:opentelemetry-javaagent\` have the \`-alpha\` suffix attached to their version number, reflecting that they are still alpha quality and will continue to have breaking changes. Please see the [VERSIONING.md](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/VERSIONING.md#opentelemetry-java-instrumentation-versioning) for more details.
```
Comment (jack-berg) [inline .github/workflows/release.yml:132]: This doesn't apply here does it?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -146,35 +165,125 @@ jobs:
...
         run: |
-          cp jmx-metrics/build/libs/opentelemetry-jmx-metrics-$VERSION-alpha.jar opentelemetry-jmx-metrics.jar
+          cp javaagent/build/libs/opentelemetry-javaagent-${VERSION}.jar opentelemetry-javaagent.jar
           gh release create --target $GITHUB_REF_NAME \
                             --title "Version $VERSION" \
                             --notes-file /tmp/release-notes.txt \
                             --discussion-category announcements \
                             v$VERSION \
-                            opentelemetry-jmx-metrics.jar
+                            opentelemetry-javaagent.jar
```
Comment (jack-berg) [inline .github/workflows/release.yml:174]: This doesn't seem right.

**Review by jack-berg — COMMENTED** [review]
Might have noticed a couple of copy / paste errors

File: .github/workflows/release.yml (hunk 1 above)
Comment (trask) [reply .github/workflows/release.yml]: oops!

File: .github/workflows/release.yml (hunk 2 above)
Comment (trask) [reply .github/workflows/release.yml]: indeed it is not!

**Review by jack-berg — APPROVED** [review]

**Pull request merged**